vscode/resources/sample-provider-settings.yaml

```diff
@@ -0,0 +1,40 @@
+---
+# This is a sample settings file with a few common providers and models defined
+
+environment:
+  ALWAYS_APPLIED_KEY: "envvar to be set regardless of which model is active"
+
+models:
+  OpenAI: &active
```

Comment:
IMHO it's more natural to store the active model in the standard settings. If we choose to store it here then we should provide a way to show that in the settings.

Reply:
The env vars needed for a model vary enough that keeping the configs all in one file makes the most sense to me.  Feels better to keep everything needed to configure the model in one place.

If not, then the yaml file should really just be a env file and the rest of the model config arguments would be normal settings.json configurations.

Reply:
My point that user should be able to check what model is being used in the settings. Actually this should be the most important and visible setting. The details of the models might be configured it in the external file (one ore many). 
The other point is - what happens if config file is shared between workspaces?

Reply:
I have to agree with @sjd78 that having these values (models in a provider-settings.yaml and "which model?" setting in settings.json) would be more confusing than having them together.

> Actually this should be the most important and visible setting.

I'm not certain I fully grok what you mean here. I would argue this could be accomplished by the user knowing **exactly** where that value is defined. I imagine a scenario where a user has no configuration, he must first define the provider in the provider-settings.yaml and then he must declare he wants to use that provider in settings.json?

> What happens if config file is shared between workspaces?

Assuming you mean the provider-settings.yaml, it is using the globalStorageUri (line 55 in paths.ts).

Reply:
For 0.0.5, we'll leave the file and handling as-is.

Going forward, there are 2 things that could be added:
  1. Have a substitution from the config file to protected storage so an api key doesn't live as clear text at rest.
  2. Add a new config key to the extension's configuration allowing selection of either the "active" anchor/link block or a named block from the config file to allow selection of the model via the settings.json config file.

Reply:
I'm OK with leaving it as-it-is  - those concerned about privacy may still provide secrets as env var the usual way.